components/dashboard/src/admin/UserDetail.tsx

```diff
@@ -290,8 +362,92 @@ export default function UserDetail(p: { user: User }) {
+                onEnter={() => false}
+                buttons={[
+                    <button
+                        className="secondary"
+                        disabled={usageLimit === costCenter?.spendingLimit}
+                        onClick={async () => {
+                            if (usageLimit !== undefined) {
+                                await getGitpodService().server.adminSetUsageLimit(
+                                    AttributionId.render(AttributionId.create(user)),
+                                    usageLimit || 0,
+                                );
+                                setUsageLimit(undefined);
+                                initialize();
+                                setEditSpendingLimit(false);
+                            }
+                        }}
+                    >
+                        Save
+                    </button>,
+                ]}
+            >
+                <p>Update the spending limit for this account.</p>
+                <div className="flex flex-col">
+                    <input
+                        type="number"
+                        min={usage?.creditsUsed}
+                        max={500000}
+                        title="Update Usage Limit"
+                        value={usageLimit}
+                        onChange={(event) => setUsageLimit(Number.parseInt(event.target.value))}
+                    />
+                </div>
+            </Modal>
+            <Modal
+                onEnter={() => false}
+                visible={editAddCreditNote}
+                onClose={() => setEditAddCreditNote(false)}
+                title="Add Credits"
+                buttons={[
+                    <button
+                        className="secondary"
+                        disabled={creditNote.credits === 0 || !creditNote.note}
+                        onClick={async () => {
+                            if (creditNote.credits !== 0 && !!creditNote.note) {
+                                await getGitpodService().server.adminAddUsageCreditNote(
+                                    AttributionId.render(AttributionId.create(user)),
+                                    creditNote.credits,
+                                    creditNote.note,
+                                );
+                                setEditAddCreditNote(false);
+                                setCreditNote({ credits: 0 });
+                                initialize();
+                            }
+                        }}
+                    >
+                        Save
+                    </button>,
+                ]}
+            >
+                <p>Adds or subtracts the amount of credits from this account.</p>
+                <div className="flex flex-col">
+                    <h4 className="mt-4">Credits</h4>
+                    <input
+                        type="number"
+                        min={-50000}
+                        max={50000}
```

Comment:
[nit] Feels arbitrary, and if we want to enforce it, perhaps should do that on the backend anyway. I'd remove.